Tests/Engine/Settings.tests.ps1

```diff
@@ -37,10 +37,13 @@ Describe "Settings Class" {
             $settings = New-Object -TypeName $settingsTypeName -ArgumentList @{}
         }
 
-        'IncludeRules', 'ExcludeRules', 'Severity', 'RuleArguments' | ForEach-Object {
-            It ("Should return empty {0} property" -f $_) {
-                $settings.${$_}.Count | Should -Be 0
-            }
+        It "Should return empty <name> property" -TestCases @(
+            @{ Name = "IncludeRules" }
+            @{ Name = "ExcludeRules" }
+            @{ Name = "Severity" }
+            @{ Name = "RuleArguments" }
+        ) {
```

Comment:
this needs a param statement to work, i believe
```
{
param ( $name )
${settings}.${Name}.Count | Should -Be 0
}
```

Reply:
yes, you are right. thanks for pointing it out.

Reply:
it's because property references don't provide null reference exceptions unless you're in strict mode. `[int]$null.count` will return 0 (`$null.count` will be null, and `[int]$null` is `0`), but `$null.GetType()` will throw because it's a method invocation.
```
PS> Set-StrictMode -Version 2                                                                                                               
PS> $null.count                                                                                                                             
The property 'count' cannot be found on this object. Verify that the property exists.
At line:1 char:1
+ $null.count
+ ~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
+ FullyQualifiedErrorId : PropertyNotFoundStrict
```